### ISSUE TITLE
QWEN2 VL forward function extremely slow for videos as short as 1.5s 

### ISSUE THREAD

**Issue by snpushpi** [QwenLM/Qwen2-VL#265]
`
from transformers import Qwen2VLForConditionalGeneration, AutoTokenizer, AutoProcessor
from qwen_vl_utils import process_vision_info
import torch 

# default: Load the model on the available device(s)
#model = Qwen2VLForConditionalGeneration.from_pretrained(
#    "Qwen/Qwen2-VL-7B-Instruct", torch_dtype="auto", device_map="auto"
#)

# We recommend enabling flash_attention_2 for better acceleration and memory saving, especially in multi-image and video scenarios.
model = Qwen2VLForConditionalGeneration.from_pretrained(
    "Qwen/Qwen2-VL-7B-Instruct",
    torch_dtype=torch.bfloat16,
    device_map="auto")

# default processer
processor = AutoProcessor.from_pretrained("Qwen/Qwen2-VL-7B-Instruct")

messages = [
    {
        "role": "user",
        "content": [
            {
                "type": "video",
                "video": "clip.mp4",
                "fps": 1.0,
            },
            {"type": "text", "text": "Describe this video."},
        ],
    }
]

# Preparation for inference
text = processor.apply_chat_template(
    messages, tokenize=False, add_generation_prompt=True
)
image_inputs, video_inputs = process_vision_info(messages)
inputs = processor(
    text=[text],
    images=image_inputs,
    videos=video_inputs,
    padding=True,
    return_tensors="pt",
)
#nputs = inputs.to("cuda")
#Takes forever in the forward function
generated_ids = model(**inputs, output_hidden_states=True)
`

I was suggested to use `qwen-vl-utils[decord]` and 'qwen-vl-utils using decord to read video.' is also printed in the line before the last line. Why does it take so long? The video i am trying is literally a video of 1.5 seconds, so it's literally processing about 2 frames. This doesn't make any sense to me

**Comment by jackbravo** [comment]
I've found this code to produce better results, since it ends up using MPS instead of CPU (which seems to be the default when you don't specify it for inputs):

So, special code changes are here (device_map="auto")

```python
model = Qwen2VLForConditionalGeneration.from_pretrained(
    "Qwen/Qwen2-VL-2B-Instruct", torch_dtype="auto", device_map="auto"
)
```

and here

```python
inputs = processor(
  text=[text],
  images=image_inputs,
  videos=video_inputs,
  padding=True,
  return_tensors="pt",
)
inputs = inputs.to("mps") # <- without this it errors out with CPU and MPS incompatibility errors
```

It is still slow on my machine, 2 minutes to generate output for an image, but previously I was cancelling processes of over 15 minutes with no output given.

**Comment by jbohnslav** [comment]
decord is a dead project with an unsolved memory leak... I'd beware of relying on it.

**Comment by davidluciolu** [comment]
@jbohnslav hello, I'm also facing the memory leak problem, any suggestions for this?
